src/app/fddev/quic_trace/fd_quic_trace_rx_tile.c

```diff
@@ -0,0 +1,159 @@
+  if( !conn_entry ) return;
+  fd_quic_conn_t *        conn = translate_ptr( fd_quic_trace_ctx_remote, conn_entry->conn );
+  fd_quic_crypto_keys_t * keys = &conn->keys[ fd_quic_enc_level_appdata_id ][ 0 ];
+
+  ulong pkt_number_off = 9UL;
+  int hdr_err = fd_quic_crypto_decrypt_hdr( data, data_sz, pkt_number_off, keys );
+  if( hdr_err!=FD_QUIC_SUCCESS ) return;
+
+  ulong pkt_number = fd_quic_pktnum_decode( data+9UL, fd_quic_h0_pkt_num_len( data[0] )+1u );
+  int crypt_err = fd_quic_crypto_decrypt( data, data_sz, pkt_number_off, pkt_number, keys );
+  if( crypt_err!=FD_QUIC_SUCCESS ) return;
+
+  static FD_TL ulong trace;
+  if( trace++ % 4096 == 0 ) {
+    FD_LOG_NOTICE(( "ip4=" FD_IP4_ADDR_FMT " conn=%016lx", FD_IP4_ADDR_FMT_ARGS( ip4_saddr ), dst_conn_id ));
+  }
+
+  (void)ip4_saddr; (void)conn;
+}
+
+static void
+fd_quic_trace_pkt( fd_quic_ctx_t * ctx,
+                   uchar *         data,
+                   ulong           data_sz,
+                   uint            ip4_saddr ) {
+  /* FIXME: for now, only handle 1-RTT */
+  int is_long = fd_quic_h0_hdr_form( data[0] );
+  if( is_long ) return;
+  fd_quic_trace_1rtt( ctx, data, data_sz, ip4_saddr );
+}
+
+static void
+after_frag( void * _ctx FD_FN_UNUSED,
+            ulong  in_idx,
+            ulong  seq,
+            ulong  sig,
+            ulong  chunk,
+            ulong  sz,
+            ulong  tsorig,
+            fd_stem_context_t * stem ) {
+  (void)in_idx; (void)seq; (void)sig; (void)chunk; (void)sz; (void)tsorig; (void)stem;
+
+  fd_quic_ctx_t * ctx = &fd_quic_trace_ctx;
+
+  if( sz < FD_QUIC_SHORTEST_PKT ) return;
+  if( sz > sizeof(ctx->buffer)  ) return;
+
+  uchar * cur  = ctx->buffer;
+  uchar * end  = cur+sz;
+
+  fd_eth_hdr_t const * eth_hdr = fd_type_pun_const( cur );
+  cur += sizeof(fd_eth_hdr_t);
+  if( FD_UNLIKELY( cur>end ) ) return;
+  if( FD_UNLIKELY( fd_ushort_bswap( eth_hdr->net_type )!=FD_ETH_HDR_TYPE_IP ) ) return;
+
+  fd_ip4_hdr_t const * ip4_hdr = fd_type_pun_const( cur );
+  if( FD_UNLIKELY( cur+sizeof(fd_ip4_hdr_t) > end ) ) return;
+  cur += FD_IP4_GET_LEN( *ip4_hdr );
+  if( FD_UNLIKELY( cur>end ) ) return;
+  if( FD_UNLIKELY( ip4_hdr->protocol!=FD_IP4_HDR_PROTOCOL_UDP ) ) return;
+
+  fd_udp_hdr_t const * udp_hdr = fd_type_pun_const( cur );
+  if( FD_UNLIKELY( cur+sizeof(fd_udp_hdr_t) > end ) ) return;
+  cur += sizeof(fd_udp_hdr_t);
+  if( FD_UNLIKELY( cur>end ) ) return;
+  (void)udp_hdr;
+
+  uint ip4_saddr = fd_uint_load_4( ip4_hdr->saddr_c );
+  fd_quic_trace_pkt( ctx, cur, (ulong)( end-cur ), ip4_saddr );
+}
+
+
+#define STEM_BURST (1UL)
+#define STEM_CALLBACK_CONTEXT_TYPE  void
+#define STEM_CALLBACK_CONTEXT_ALIGN 1
+#define STEM_CALLBACK_BEFORE_FRAG   before_frag
+#define STEM_CALLBACK_DURING_FRAG   during_frag
+#define STEM_CALLBACK_AFTER_FRAG    after_frag
+#include "../../../disco/stem/fd_stem.c"
+
+fd_topo_run_tile_t fd_tile_quic_trace_rx = {
```

Comment:
I don't think you need this? Not used anywhere.

Reply:
@mmcgee-jump I'd like to keep it. I want this to work as a real time eventually